web/src/admin/AdminFrontConf.js

```diff
@@ -0,0 +1,202 @@
+// Copyright 2021 The casbin Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from "react";
+import { withRouter, Link } from "react-router-dom";
+import * as Setting from "../Setting";
+import i18next from "i18next";
+import * as ConfBackend from "../backend/ConfBackend";
+
+class AdminFrontConf extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      classes: props,
+      Management_LIST: [
+        { label: "Visual Conf", value: "visualConf" },
+      ],
+      field: "visualConf",
+      conf: [],
+      form: {},
+      message: "",
+      memberId: props.match.params.memberId,
+    };
+    this.state.url = `/admin/frontconf`;
+  }
+
+  componentDidMount() {
+    this.getFrontConf(this.state.field);
+  }
+
+  changeField(field) {
+    this.setState({
+        field: field,
+      }, ()=>{
+        this.getFrontConf(this.state.field)
+      }
+    );
+  }
+
+  renderManagementList(item) {
+    return (
+        <a
+            href="javascript:void(0);"
+            className={this.state.field === item.value ? "tab_current" : "tab"}
+            onClick={() => this.changeField(item.value)}
+        >
+          {item.label}
+        </a>
+    );
+  }
+
+  getFrontConf(field) {
+    ConfBackend.getFrontConfByField(field).then((res) => {
+      let form = this.state.form;
+      for (var k in res) {
+        form[res[k].Id] = res[k].Value;
+      }
+      this.setState({
+        conf: res
+      })
+    });
+  }
+
+  clearMessage() {
+    this.setState({
+      message: "",
+    });
+  }
+
+  inputChange(event,id) {
+    let form = this.state.form;
+    form[id] = event.target.value;
+    this.setState({
+      form: form,
+    });
+  }
+
+  updateConf() {
+    let confs = [];
+    for (var k in this.state.form) {
+      confs.push({Id:k,Value:this.state.form[k]})
```

Comment:
Because xorm does not support batch updates, we need to update multiple times in the backend, so we need to minimize update entries as much as possible. 
And this will update all records.